swank/abcl.lisp

```diff
@@ -541,12 +541,19 @@
 (defmethod source-location ((symbol symbol))
   (when (pathnamep (ext:source-pathname symbol))
     (let ((pos (ext:source-file-position symbol)))
-      `(:location
-        (:file ,(namestring (ext:source-pathname symbol)))
-        ,(if pos
-             (list :position (1+ pos))
-             (list :function-name (string symbol)))
-        (:align t)))))
+      (if (ext:pathname-jar-p (ext:source-pathname symbol))
+          (destructuring-bind (jar entry) (abcl-asdf::split-string (subseq (namestring (ext:source-pathname symbol)) 9) (char "!" 0))
```

Comment:
There's a `split-string` function in this file. Why are you grabbing it from abcl-asdf? Also, what does the `(subseq ... 9)` do? (i.e. what's it discarding?) Perhaps a comment or a clearer way of doing this (perhaps some pathname manipulation?) would be better?